glommio/src/task/task_impl.rs

```diff
@@ -125,74 +118,9 @@ impl<T> Task<T> {
 
         unsafe { ((*header).vtable.run)(ptr) }
     }
-
```

Comment:
wait, this feels like too much. or at least needs to be justified separately.

We can't cancel tasks now?

Reply:
Hi @glommer , the task can be cancelled using JoinHandle. Here is the code from the Task API.

```
pub async fn cancel(self) -> Option<T> {
        self.0.cancel().await
    }
```

And down to the stack of calls to the JoinHandle.

Reply:
Thanks.

That makes sense. In the future, I expect these kind of things to feature in the commit message, to help
both the reviewer and future readers understand the change.

I consider you a part of the family now,  so I'll be a bit more strict on this =)

for now I am merging this